src/histogram.ts

```diff
@@ -18,82 +18,112 @@ import { SummaryValue } from "@dynatrace/metric-utils";
 import { DataPoint, Histogram } from "@opentelemetry/sdk-metrics-base";
 
 export function estimateHistogram(point: DataPoint<Histogram>): SummaryValue {
-	if (point.value.count === 0) {
-		return { count: 0, sum: 0, min: 0, max: 0 };
-	}
-
-	const { min, max } = getMinMax(point);
+	const { min, max, sum } = estimateOptionalProperties(point);
 
 	return {
 		count: point.value.count,
-		sum: point.value.sum,
+		sum,
 		max,
 		min
 	};
 }
 
-function getMinMax(point: DataPoint<Histogram>): { min: number; max: number } {
-	if (point.value.hasMinMax) {
-		return { min: point.value.min, max: point.value.max };
-	}
+function estimateSingleBucketHistogram(point: DataPoint<Histogram>): { min: number; max: number; sum: number } {
+	const sum = point.value.sum ?? 0;
+	const mean = sum / point.value.count;
 
-	return estimateHistMinMax(point);
+	return { min: point.value.min ?? mean, max: point.value.max ?? mean, sum };
 }
 
-// Expect to be called only with points that contain actual data (point.value.count > 0)
-function estimateHistMinMax(point: DataPoint<Histogram>): { min: number; max: number } {
-	const { counts, boundaries } = point.value.buckets;
-	const avg = point.value.sum / point.value.count;
+function estimateOptionalProperties(point: DataPoint<Histogram>): { min: number; max: number; sum: number } {
+	// shortcut if min, max, and sum are provided
+	if (point.value.min != null && point.value.max != null && point.value.sum != null) {
+		return { min: point.value.min, max: point.value.max, sum: point.value.sum };
+	}
+
+	// Shortcut for 0 count histograms
+	if (point.value.count === 0) {
```

Comment:
Can this be null? I assume no, then its fine.

Reply:
You're right, it cannot be null. :slightly_smiling_face: